neurodamus/node.py

```diff
@@ -1017,15 +1045,19 @@ def enable_reports(self):
 
             self._report_list.append(report)
 
+        if SimConfig.restore_coreneuron:
+            CoreConfig.update_report_config(substitutions)
+
```

Comment:
Can we call `update_report_config` inside `_coreneuron_write_report_config` just like the update_stop before?  To me it looks better to call creation or update in one place.

Reply:
2 reasons why I am against this approach:

- it is way less efficient. Notice that now the update is done only once, at the end. We buffer all the updates in 1 structure and apply them in 1 go. In the other way, if we have multiple reports, we load the file, look for the tstop line, change and rewrite the whole file. We read/write the whole file to change just one line, multiple times
- it is not SOLID: updating and writing are different jobs

Reply:
done